[BUG] VS Code / Metals / Bloop integration fails with java.lang.RuntimeException: 'boom'

**Describe the bug**
```
ERROR Unexpected error when compiling rapids-4-spark-sql_2.12: 'java.lang.RuntimeException: boom'
```
**Steps/Code to reproduce bug**

Create Bloop build targets
```
mvn clean install ch.epfl.scala:maven-bloop_2.13:bloopInstall -pl tests -am -Dskip -DskipTests -Dmaven.javadoc.skip -Dbuildver=331 -DdownloadSources=true
```

Start Metals and observe in Metals output log:
```
2023.02.13 21:53:27 ERROR Unexpected error when compiling rapids-4-spark-sql_2.12: 'java.lang.RuntimeException: boom'
```

Connecting to Bloop server from the CLI client reveals a more clear stacktrace:
```
Caused by: java.lang.RuntimeException: boom
        at com.sourcegraph.semanticdb_javac.SemanticdbTaskListener.finished(SemanticdbTaskListener.java:76)
        at com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
        at com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
        at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
        at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
        at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
        at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
        at com.sun.tools.javac.main.Main.compile(Main.java:523)
        ... 24 more
Caused by: com.sourcegraph.semanticdb_javac.CompilationUnitException: file:/home/user/gits/NVIDIA/spark-rapids/sql-plugin/src/main/java/com/nvidia/spark/rapids/ColumnarCopyHelper.java
Caused by: com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.apache.spark.sql.types.Decimal$1 not found
```

https://github.com/sourcegraph/scip-java/blob/b7d268233f1a303f66b6d9804a68f64b1e5d7032/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java#L76

**Expected behavior**

Metals should continue to work 

**Environment details (please complete the following information)**
 - Environment location: local dev

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -264,7 +264,7 @@ interested in. For example, to generate the Bloop projects for the Spark 3.2.0 d
 just for the production code run:
 
 ```shell script
-mvn install ch.epfl.scala:maven-bloop_2.13:1.4.9:bloopInstall -pl aggregator -am \
+mvn install ch.epfl.scala:bloop-maven-plugin:bloopInstall -pl aggregator -am \
   -DdownloadSources=true \
   -Dbuildver=320 \
   -DskipTests \
@@ -296,7 +296,7 @@ You can now open the spark-rapids as a
 
 Read on for VS Code Scala Metals instructions.
 
-# Bloop, Scala Metals, and Visual Studio Code
+#### Bloop, Scala Metals, and Visual Studio Code
 
 _Last tested with 1.63.0-insider (Universal) Commit: bedf867b5b02c1c800fbaf4d6ce09cefba_
 
@@ -337,6 +337,26 @@ jps -l
 72356 bloop.Server
 72349 scala.meta.metals.Main
 ```
+
+##### Known Issues
+
+###### java.lang.RuntimeException: boom`
+
+Metals background compilation process status appears to be resetting to 0% after reaching 99%
+and you see a peculiar error message [`java.lang.RuntimeException: boom`][1]. You can work around
+it by making sure Metals Server, Bloop Server are running on Java 11+.
+
+1. To this end make sure that Bloop projects are generated using Java 11+
+
+    ```bash
+    JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64 \
+      mvn install ch.epfl.scala:bloop-maven-plugin:bloopInstall \
+      -DdownloadSources=true \
+      -Dbuildver=331 \
+      -Dskip -DskipTests -Dmaven.javadoc.skip
+    ```
+
+1. Add `metals.javaHome` to VSCode preferences to point to Java 11+.
 
 #### Other IDEs
 We welcome pull requests with tips how to setup your favorite IDE!
@@ -514,3 +534,5 @@ Options:
 
 ## Attribution
 Portions adopted from https://github.com/rapidsai/cudf/blob/main/CONTRIBUTING.md, https://github.com/NVIDIA/nvidia-docker/blob/master/CONTRIBUTING.md, and https://github.com/NVIDIA/DALI/blob/main/CONTRIBUTING.md
+
+[1]: https://github.com/sourcegraph/scip-java/blob/b7d268233f1a303f66b6d9804a68f64b1e5d7032/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java#L76

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -340,11 +340,11 @@ jps -l
 
 ##### Known Issues
 
-###### java.lang.RuntimeException: boom`
+###### java.lang.RuntimeException: boom
 
 Metals background compilation process status appears to be resetting to 0% after reaching 99%
 and you see a peculiar error message [`java.lang.RuntimeException: boom`][1]. You can work around
-it by making sure Metals Server, Bloop Server are running on Java 11+.
+it by making sure Metals Server (Bloop client) and Bloop Server are both running on Java 11+.
 
 1. To this end make sure that Bloop projects are generated using Java 11+
 
@@ -356,8 +356,11 @@ it by making sure Metals Server, Bloop Server are running on Java 11+.
       -Dskip -DskipTests -Dmaven.javadoc.skip
     ```
 
-1. Add `metals.javaHome` to VSCode preferences to point to Java 11+.
+1. Add [`metals.javaHome`][2] to VSCode preferences to point to Java 11+.
 
+[1]: https://github.com/sourcegraph/scip-java/blob/b7d268233f1a303f66b6d9804a68f64b1e5d7032/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java#L76
+
+[2]: https://github.com/scalameta/metals-vscode/pull/644/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626R132
 #### Other IDEs
 We welcome pull requests with tips how to setup your favorite IDE!
 
@@ -534,5 +537,3 @@ Options:
 
 ## Attribution
 Portions adopted from https://github.com/rapidsai/cudf/blob/main/CONTRIBUTING.md, https://github.com/NVIDIA/nvidia-docker/blob/master/CONTRIBUTING.md, and https://github.com/NVIDIA/DALI/blob/main/CONTRIBUTING.md
-
-[1]: https://github.com/sourcegraph/scip-java/blob/b7d268233f1a303f66b6d9804a68f64b1e5d7032/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java#L76

--- a/build/buildall
+++ b/build/buildall
@@ -19,8 +19,6 @@ set -e
 
 shopt -s extglob
 
-BLOOP_VERSION=${BLOOP_VERSION:-"1.4.13"}
-BLOOP_SCALA_VERSION=${BLOOP_SCALA_VERSION:-"2.13"}
 SKIP_CLEAN=1
 BUILD_ALL_DEBUG=0
 
@@ -66,7 +64,7 @@ function bloopInstall() {
       bloopTmpConfigDir="$bloopTmpDir/.bloop$bv"
       mkdir -p $bloopTmpConfigDir
       $MVN -B clean install \
-        ch.epfl.scala:maven-bloop_${BLOOP_SCALA_VERSION}:${BLOOP_VERSION}:bloopInstall \
+        ch.epfl.scala:bloop-maven-plugin:bloopInstall \
         -pl aggregator -am \
         -Dbloop.configDirectory="$bloopTmpConfigDir" \
         -DdownloadSources=true \

--- a/build/buildall
+++ b/build/buildall
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright (c) 2021-2022, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2021-2023, NVIDIA CORPORATION. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/jenkins/databricks/build.sh
+++ b/jenkins/databricks/build.sh
@@ -337,7 +337,7 @@ else
 fi
 
 if [[ "$WITH_BLOOP" == "1" ]]; then
-    MVN_OPT="ch.epfl.scala:maven-bloop_2.13:bloopInstall $MVN_OPT"
+    MVN_OPT="ch.epfl.scala:bloop-maven-plugin:bloopInstall $MVN_OPT"
 fi
 
 # Build the RAPIDS plugin by running package command for databricks

